WooCommerce/src/main/kotlin/com/woocommerce/android/ui/searchfilter/SearchFilterAdapter.kt

```diff
@@ -0,0 +1,65 @@
+package com.woocommerce.android.ui.searchfilter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import com.woocommerce.android.databinding.FilterListItemBinding
+import com.woocommerce.android.extensions.hide
+import com.woocommerce.android.ui.searchfilter.SearchFilterAdapter.SearchFilterViewHolder
+
+class SearchFilterAdapter(
+    private val onItemSelectedListener: (SearchFilterItem) -> Unit
+) : RecyclerView.Adapter<SearchFilterViewHolder>() {
```

Comment:
💡  If we use `DiffUtil`, maybe it makes sense to use [`ListAdapter`](https://developer.android.com/reference/androidx/recyclerview/widget/ListAdapter) class? It computes diff on background thread and make codebase consistent. We already use it in e.g. `ProductFilterOptionListAdapter`. WDYT?